wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,131 @@
+			return;
+		}
+
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+
+		// Access the subtype
+		$subtype = $decodedData['Eck_Collection_Camp1'][0]['fields']['subtype'] ?? null;
+
+		// Check if the subtype is 4 or 5
+		if ($subtype != 4 && $subtype != 5) {
+			return;
+		}
+
+		// Access the id within the decoded data
+		if (!isset($decodedData['Eck_Collection_Camp1'][0]['fields']['id'])) {
+			return;
+		}
+
+		$campId = $decodedData['Eck_Collection_Camp1'][0]['fields']['id'];
+
+		// Fetch the collection camp details
+		$collectionCamps = \Civi\Api4\EckEntity::get('Collection_Camp', FALSE)
+			->addWhere('id', '=', $campId)
+			->setLimit(1)
+			->execute();
+
+		if (empty($collectionCamps)) {
+			return;
+		}
+
+		$collectionCampsCreatedDate = $collectionCamps->first()['created_date'] ?? null;
+
+		// Get the year
+		$year = date('Y', strtotime($collectionCampsCreatedDate));
+
+		// Fetch the state ID from the collection camp intent details
+		$stateId = $decodedData['Eck_Collection_Camp1'][0]['fields']['Collection_Camp_Intent_Details.State'] ?? null;
+		if ($subtype == 5) {
+			$stateId = $decodedData['Eck_Collection_Camp1'][0]['fields']['Dropping_Centre.State'] ?? null;
+		}
+
+		if (!$stateId) {
+			return;
+		}
+
+		// Fetch the state abbreviation
+		$stateProvinces = \Civi\Api4\StateProvince::get(TRUE)
+			->addWhere('id', '=', $stateId)
+			->setLimit(1)
+			->execute();
+
+		if (empty($stateProvinces)) {
+			return;
+		}
+
+		$stateAbbreviation = $stateProvinces->first()['abbreviation'] ?? null;
+
+		if (!$stateAbbreviation) {
+			return;
+		}
+
+		// Fetch the Goonj-specific state code
+		$goonjStateCodePath = ABSPATH . 'wp-content/civi-extensions/goonjcustom/config/constants.php';
+		$goonjStateCode = include $goonjStateCodePath;
+		$stateCode = $goonjStateCode[$stateAbbreviation] ?? 'UNKNOWN';
+
+		// Get the current event title
+		$currentTitle = $decodedData['Eck_Collection_Camp1'][0]['fields']['title'] ?? 'Collection Camp';
+
+		// Fetch the event code
+		$eventCodePath = ABSPATH . 'wp-content/civi-extensions/goonjcustom/config/eventCode.php';
+		$eventCodeConfig = include $eventCodePath;
+
+		// Determine the event code based on the current title
+		$eventCode = $eventCodeConfig[$currentTitle] ?? 'UNKNOWN';
+
+		// Count existing camps for the state and year with the same event code
+		$existingCamps = \Civi\Api4\EckEntity::get('Collection_Camp', FALSE)
+			->addSelect('title')
+			->addWhere('title', 'LIKE', "$year/$stateCode/$eventCode/%")
+			->execute();
+
+		$serialNumber = sprintf('%03d', $existingCamps->count() + 1);
+
+		// Modify the title to include the year, state code, event code, and serial number
+		$newTitle = $year . '/' . $stateCode . '/' . $eventCode . '/' . $serialNumber;
+		$decodedData['Eck_Collection_Camp1'][0]['fields']['title'] = $newTitle;
+
+		// Update the objectRef's data to reflect the new title
+		$objectRef->data = json_encode($decodedData);
```

Comment:
What's the purpose of doing this?

Reply:
we don't need this, i think I forgot to remove what I was testing, Let me check on other places too and remove